### PR TITLE
Add Documentation and build workflow using CI

### LINKED ISSUE #2: Add planar non-periodic example to README.md
Add example using a MALI mesh to the README.md

### PR DESCRIPTION
**Pull request by andrewdnolan**
This PR adds some basic documentation building off what already existed in the `README.md`. The documentation now includes: 
- installation instruction using `conda-forge`*
- global mesh example
- planar non-periodic example*
- developer installation instructions 
- basic api documentation*  

where stared bullets have been added as part of this PR. The documentation is build through GitHub actions, following the workflow from `mache`.  This PR closes #2.

### CODE REVIEW

**Comment by andrewdnolan** [comment]
While working on this branch, I had the documentation deployed on a push to `docs`. I've changed that to `main` for the review/finalization process, but the documentation can be seen here: 
https://docs.e3sm.org/mosaic/

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'mosaic'
+copyright = '2024, E3SM Development Team'
+author = 'E3SM Development Team'
+release = '0.1.0'
```
[the remark this answers is not in the document]
Comment (andrewdnolan) [reply docs/conf.py]: I've added a `version` module, where the `__version__` can be parsed dynamically. Thanks for suggestion!

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    'myst_parser',
+    'sphinx.ext.autodoc',
+    'sphinx.ext.autosummary',
+]
+
+autosummary_generate = ['developers_guide/api.md']
+
+templates_path = ['_templates']
+
+exclude_patterns = ["_build", ".DS_Store"]
```
[the remark this answers is not in the document]
Comment (andrewdnolan) [reply docs/conf.py:30]: I've added the `intersphinx` extension, and update what existed in the docstrings to use cross referencing. The current docstrings are far from complete, but for the sake of getting documentation with `conda-forge` install instructions out I might opt to improve those as part of a future PR.

**Review by xylar — COMMENTED** [review]
A few more suggested fixed to the docstrings.

**Comment by andrewdnolan** [comment]
Thanks for the suggestions! Just built the documentation locally with those two most recent commits and everything seems ok to me.

**Pull request merged**